src/yuzu/main.cpp

```diff
@@ -2840,170 +2840,350 @@ void GMainWindow::OnGameListNavigateToGamedbEntry(u64 program_id,
+#elif defined(_WIN32)
+    shortcut_path_full += ".lnk";
+#endif
+
+    LOG_INFO(Common, "[GMainWindow::CreateShortcutLink] Create shortcut path: {}",
+             shortcut_path_full.string());
+
+#if defined(__linux__) || defined(__FreeBSD__)
+    // This desktop file template was writing referencing
+    // https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-1.0.html
+    try {
+
+        // Plus 'Type' is required
+        if (name.empty()) {
+            LOG_ERROR(Common, "[GMainWindow::CreateShortcutLink] Name is empty");
+            shortcut_succeeded = false;
+            return shortcut_succeeded;
         }
-        UISettings::values.shortcut_already_warned = true;
-    }
-#endif // __linux__
+        std::ofstream shortcut_stream(shortcut_path_full, std::ios::binary | std::ios::trunc);
 
-    std::filesystem::path target_directory{};
+        if (shortcut_stream.is_open()) {
 
-    switch (target) {
-    case GameListShortcutTarget::Desktop: {
-        const QString desktop_path =
-            QStandardPaths::writableLocation(QStandardPaths::DesktopLocation);
-        target_directory = desktop_path.toUtf8().toStdString();
-        break;
-    }
-    case GameListShortcutTarget::Applications: {
-        const QString applications_path =
-            QStandardPaths::writableLocation(QStandardPaths::ApplicationsLocation);
-        if (applications_path.isEmpty()) {
-            const char* home = std::getenv("HOME");
-            if (home != nullptr) {
-                target_directory = std::filesystem::path(home) / ".local/share/applications";
+            fmt::print(shortcut_stream, "[Desktop Entry]\n");
+            fmt::print(shortcut_stream, "Type=Application\n");
+            fmt::print(shortcut_stream, "Version=1.0\n");
+            fmt::print(shortcut_stream, "Name={}\n", name);
+
+            if (!comment.empty()) {
+                fmt::print(shortcut_stream, "Comment={}\n", comment);
             }
+
+            if (std::filesystem::is_regular_file(icon_path)) {
+                fmt::print(shortcut_stream, "Icon={}\n", icon_path.string());
+            }
+
+            fmt::print(shortcut_stream, "TryExec={}\n", command.string());
+            fmt::print(shortcut_stream, "Exec={}", command.string());
+
+            if (!arguments.empty()) {
+                fmt::print(shortcut_stream, " {}", arguments);
+            }
+
+            fmt::print(shortcut_stream, "\n");
+
+            if (!categories.empty()) {
+                fmt::print(shortcut_stream, "Categories={}\n", categories);
+            }
+
+            if (!keywords.empty()) {
+                fmt::print(shortcut_stream, "Keywords={}\n", keywords);
+            }
+
+            shortcut_stream.close();
+            return true;
+
         } else {
-            target_directory = applications_path.toUtf8().toStdString();
+            LOG_ERROR(Common, "[GMainWindow::CreateShortcutLink] Failed to create shortcut");
```

Comment:
this is not how we do logging in yuzu, I suggested taking a look at other logging functions to see how to do it properly.
```suggestion
            LOG_ERROR(Frontend, "Failed to create shortcut");
```

Reply:
I consider it a good way to debug the code but I can delete it.